libgadget/slotsmanager.c

```diff
@@ -22,39 +22,86 @@ slots_gc_base();
 static int
 slots_gc_slots(int * compact_slots);
 
-/*Initialise a new slot for the particle at index i.*/
+/* Initialise a new slot with type at index pi
+ * for the particle at index i.*/
```

Comment:
/* This will modify both P[i] and the slot at pi for the ptype. */

Reply:
Done